reposerver/repository/repository.go

```diff
@@ -2573,6 +2573,25 @@ func (s *Service) GetGitFiles(_ context.Context, request *apiclient.GitFilesRequ
 		return nil, status.Errorf(codes.Internal, "unable to resolve git revision %s: %v", revision, err)
 	}
 
+	if gpg.IsGPGEnabled() && request.VerifyCommit {
+		cs, err := gitClient.VerifyCommitSignature(revision)
+		if err != nil {
+			log.Errorf("error verifying signature of commit '%s' in repo '%s': %v", revision, repo.Repo, err)
+			return nil, err
+		}
+
+		if cs != "" {
+			vr := gpg.ParseGitCommitVerification(cs)
+			if vr.Result == gpg.VerifyResultUnknown {
+				return nil, fmt.Errorf("UNKNOWN signature: %s", vr.Message)
+			} else {
+				log.Debugf("%s signature from %s key %s", vr.Result, vr.Cipher, gpg.KeyID(vr.KeyID))
```

Comment:
Here few more cases, like 	
VerifyResultBad     = "Bad"
VerifyResultInvalid = "Invalid", 
I think it also should be an error

Reply:
I tried keeping the cases similar to what we have today for applications to maintain the same behavior for application-sets

Reply:
Make sense for me, thanks for explanation